lib/store.js

```diff
@@ -186,11 +190,18 @@ Store.prototype.insert = function(actor, credential, options, callback) {
         {resource: credential, translate: ['issuer', 'recipient']}, callback);
     },
     insert: ['checkPermission', function(callback) {
+      var referenceIdHash;
+      if(!credential.referenceId) {
```

Comment:
Probably silly, but we may want to allow people to use the empty string as a `referenceId`. So here we should probably check `!('referenceId' in credential)` instead. We can presume that if it's present it is a string.